internal/controller/compaction/reconciler.go

```diff
@@ -126,44 +135,46 @@ func (r *Reconciler) reconcileJob(ctx context.Context, logger logr.Logger, etcd
 			// Don't need to requeue if the job is currently running
 			return ctrl.Result{}, nil
 		}
-
-		// Delete job if the job succeeded
-		if job.Status.Succeeded > 0 {
-			metricJobsCurrent.With(prometheus.Labels{druidmetrics.EtcdNamespace: etcd.Namespace}).Set(0)
-			if job.Status.CompletionTime != nil {
-				metricJobDurationSeconds.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededTrue, druidmetrics.EtcdNamespace: etcd.Namespace}).Observe(job.Status.CompletionTime.Time.Sub(job.Status.StartTime.Time).Seconds())
-			}
-			if err := r.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationForeground)); err != nil {
-				logger.Error(err, "Couldn't delete the successful job", "namespace", etcd.Namespace, "name", compactionJobName)
-				return ctrl.Result{
-					RequeueAfter: 10 * time.Second,
-				}, fmt.Errorf("error while deleting successful compaction job: %v", err)
+		// Set the current job count to 0 if the job is not active
+		metricJobsCurrent.With(prometheus.Labels{druidmetrics.LabelEtcdNamespace: etcd.Namespace}).Set(0)
+		isJobSuccessful, jobCompletionReason := isJobSuccessfulWithReason(job)
+		if isJobSuccessful {
+			recordSuccessfulJobMetrics(job)
+		} else {
+			if err := recordFailedJobMetrics(ctx, r, logger, job, etcd, jobCompletionReason); err != nil {
+				return ctrl.Result{RequeueAfter: 10 * time.Second}, fmt.Errorf("error while handling failed job: %w", err)
 			}
-			metricJobsTotal.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededTrue, druidmetrics.EtcdNamespace: etcd.Namespace}).Inc()
 		}
-
-		// Delete job and requeue if the job failed
-		if job.Status.Failed > 0 {
-			metricJobsCurrent.With(prometheus.Labels{druidmetrics.EtcdNamespace: etcd.Namespace}).Set(0)
-			if job.Status.StartTime != nil {
-				metricJobDurationSeconds.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededFalse, druidmetrics.EtcdNamespace: etcd.Namespace}).Observe(time.Since(job.Status.StartTime.Time).Seconds())
-			}
-			if err := r.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationForeground)); err != nil {
-				return ctrl.Result{
-					RequeueAfter: 10 * time.Second,
-				}, fmt.Errorf("error while deleting failed compaction job: %v", err)
-			}
-			metricJobsTotal.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededFalse, druidmetrics.EtcdNamespace: etcd.Namespace}).Inc()
+		// Delete the completed compaction job
+		if err := r.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationForeground)); err != nil {
+			logger.Error(err, "Couldn't delete the completed compaction job", "namespace", etcd.Namespace, "name", compactionJobName)
 			return ctrl.Result{
-				RequeueAfter: 10 * time.Second,
-			}, nil
+				RequeueAfter: 10 * time.Second}, fmt.Errorf("error while deleting the completed compaction job: %w", err)
 		}
 	}
 
+	jobMeta := &metav1.ObjectMeta{
```

Comment:
can you add a comment stating the reason to loop up the pod. This will make things clear as it was never done before and is due to addition of graceful termination seconds defined in the PodSpec of the Job.